smk/QC_1.smk

```diff
@@ -31,89 +30,83 @@ use rule QC_1_base, QC_1_rpkg from print_versions as version_*
 
 snakefile_name = print_versions.get_smk_filename()
 
-if config['raw_reads_dir'] is None:
-    print('ERROR in ', config_path, ': raw_reads_dir variable in configuration yaml file is empty. Please, complete ', config_path)
-else:
-    raw_dir = config['raw_reads_dir']
+raw_dir              = validate_required_key(config, 'raw_reads_dir')
+perc_remaining_reads = validate_required_key(config, 'perc_remaining_reads')
 
-multiplex_tech = 'TruSeq'
-if config['MULTIPLEX_TECH'] is None:
-    print('ERROR in ', config_path, ': MULTIPLEX_TECH variable in configuration yaml file is empty. Please, complete ', config_path)
-else:
-    multiplex_tech = config['MULTIPLEX_TECH']
+TRIMMOMATIC_threads  = validate_required_key(config, 'TRIMMOMATIC_threads')
+TRIMMOMATIC_memory   = validate_required_key(config, 'TRIMMOMATIC_memory')
 
-if config['TRIMMOMATIC_threads'] is None:
-    print('ERROR in ', config_path, ': TRIMMOMATIC_threads variable is empty. Please, complete ', config_path)
-elif type(config['TRIMMOMATIC_threads']) != int:
-    print('ERROR in ', config_path, ': TRIMMOMATIC_threads variable is not an integer. Please, complete ', config_path)
+TRIMMOMATIC_adaptors = validate_required_key(config, 'TRIMMOMATIC_adaptors')
+if TRIMMOMATIC_adaptors in ('Skip', 'Quality'):
+    pass
+elif not os.path.exists(TRIMMOMATIC_adaptors):
+    raise Exception(f"ERROR in {config_path}: TRIMMOMATIC_adaptors file path does not exist.")
 
```

Comment:
Could `TRIMMOMATIC_adaptors` type be `['file', str]` and location checked via `config_parser`?

Reply:
Please see above.